src/jrd/replication/Publisher.cpp

```diff
@@ -560,11 +560,12 @@ void REPL_erase(thread_db* tdbb, const record_param* rpb, jrd_tra* transaction)
 
 	// This temporary auto-pointer is just to delete a temporary record
 	AutoPtr<Record> cleanupRecord(record != rpb->rpb_record ? record : NULL);
+	AutoSetRestoreFlag<ULONG> noRecursion(&tdbb->tdbb_flags, TDBB_repl_sql, true);
 
```

Comment:
What is the problem you're solving here? Plugin modifying tables/generators inside the database? Should we really consider this appropriate?

Reply:
We cannot strictly prohibit it and the prevention is cheap so why not?

Reply:
OK, but please rename the flag then. TDBB_replicating or TDBB_repl_in_progress or something like that.